[ERROR DRT-0155] Guide in net net7 uses layer M6 (12) that is outside the allowed routing range [M2 (4), (M5)].

Following merge of https://github.com/The-OpenROAD-Project/OpenROAD/pull/3468 and using https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/1097 (not merged yet), I get the error below.

Create a `settings.mk` file in the flow folder and use the updated mock-array-big https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/1097:

```
export MOCK_ARRAY_PITCH_SCALE=20
export DESIGN_CONFIG=designs/asap7/mock-array-big/config.mk
```

Then run:

```
rm -rf results/ objects/ logs/ && make
```

Fails after 1 minute or so with output:

```
[deleted]
#macroValidPlanarAp    = 0
#macroValidViaAp       = 0
#macroNoAp             = 0
[INFO DRT-0166] Complete pin access.
[INFO DRT-0267] cpu time = 00:00:01, elapsed time = 00:00:00, memory = 375.52 (MB), peak = 425.54 (MB)
[ERROR DRT-0155] Guide in net net7 uses layer M6 (12) that is outside the allowed routing range [M2 (4), (M5)].
Error: detail_route.tcl, 59 DRT-0155
Command exited with non-zero status 1
Elapsed time: 0:01.72[h:]min:sec. CPU time: user 2.81 sys 0.09 (168%). Peak memory: 435752KB.
make[1]: *** [Makefile:622: results/asap7/mock-array-big_Element/base/5_2_route.odb] Error 1
make[1]: Leaving directory '/home/oyvind/OpenROAD-flow-scripts/flow'
make: *** [Makefile:329: results/asap7/mock-array-big_Element/base/Element.lib] Error 2
```

_Originally posted by @oharboe in https://github.com/The-OpenROAD-Project/OpenROAD/issues/3468#issuecomment-1593770094_

--- a/flow/designs/asap7/mock-array-big/Element/config.mk
+++ b/flow/designs/asap7/mock-array-big/Element/config.mk
@@ -36,3 +36,8 @@ export PDN_TCL = designs/asap7/mock-array-big/Element/pdn.tcl
 # At time of adding this option, only 3 iterations were needed for 0
 # violations.
 export DETAILED_ROUTE_ARGS=-bottom_routing_layer M2 -top_routing_layer M5 -save_guide_updates -verbose 1 -droute_end_iter 10
+
+# since we are specifying DETAILED_ROUTE_ARGS, we need to communicate the
+# same information to other stages in the flow.
+export MIN_ROUTING_LAYER = M2
+export MAX_ROUTING_LAYER = M5

--- a/flow/designs/asap7/mock-array-big/config.mk
+++ b/flow/designs/asap7/mock-array-big/config.mk
@@ -45,3 +45,8 @@ verilog:
 # At time of adding this option, only 12 iterations were needed for 0
 # violations.
 export DETAILED_ROUTE_ARGS=-bottom_routing_layer M2 -top_routing_layer M7 -save_guide_updates -verbose 1 -droute_end_iter 15
+
+# since we are specifying DETAILED_ROUTE_ARGS, we need to communicate the
+# same information to other stages in the flow.
+export MIN_ROUTING_LAYER = M2
+export MAX_ROUTING_LAYER = M7

